src/h5web/providers/hsds/utils.test.ts

```diff
@@ -0,0 +1,179 @@
+      const idType: HDF5Id = 'aHdfId';
+      expect(convertHsdsType(idType)).toBe(idType);
+    });
+
+    it('should left string type unchanged', () => {
+      const stringType: HDF5StringType = {
+        class: HDF5TypeClass.String,
+        charSet: 'H5T_CSET_UTF8',
+        strPad: 'H5T_STR_NULLPAD',
+        length: 25,
+      };
+      expect(convertHsdsType(stringType)).toBe(stringType);
+    });
+
+    it('should convert integer types', () => {
+      const leIntegerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      expect(convertHsdsType(leIntegerType)).toEqual({
+        class: HDF5TypeClass.Integer,
+        size: 8,
+        endianness: 'LE',
+        unsigned: undefined,
+      });
+
+      const beIntegerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_U64BE',
+      };
+      expect(convertHsdsType(beIntegerType)).toEqual({
+        class: HDF5TypeClass.Integer,
+        size: 64,
+        endianness: 'BE',
+        unsigned: true,
+      });
+    });
+
+    it('should convert float types', () => {
+      const leFloatType: HsdsBaseType = {
+        class: HDF5TypeClass.Float,
+        base: 'H5T_IEEE_F32LE',
+      };
+      expect(convertHsdsType(leFloatType)).toEqual({
+        class: HDF5TypeClass.Float,
+        size: 32,
+        endianness: 'LE',
+        unsigned: undefined,
+      });
+
+      const beFloatType: HsdsBaseType = {
+        class: HDF5TypeClass.Float,
+        base: 'H5T_IEEE_F64BE',
+      };
+      expect(convertHsdsType(beFloatType)).toEqual({
+        class: HDF5TypeClass.Float,
+        size: 64,
+        endianness: 'BE',
+        unsigned: undefined,
+      });
+    });
+
+    it('should convert the base of Array type', () => {
+      const integerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      const arrayType: HsdsArrayType = {
+        class: HDF5TypeClass.Array,
+        base: integerType,
+        dims: [4, 5],
+      };
+      expect(convertHsdsType(arrayType)).toEqual({
+        class: HDF5TypeClass.Array,
+        base: {
+          class: HDF5TypeClass.Integer,
+          size: 8,
+          endianness: 'LE',
+          unsigned: undefined,
+        },
+        dims: [4, 5],
+      });
+    });
+
+    it('should convert the base of VLen type', () => {
+      const integerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      const vlenType: HsdsVLenType = {
+        class: HDF5TypeClass.VLen,
+        base: integerType,
+      };
+      expect(convertHsdsType(vlenType)).toEqual({
+        class: HDF5TypeClass.VLen,
+        base: {
+          class: HDF5TypeClass.Integer,
+          size: 8,
+          endianness: 'LE',
+          unsigned: undefined,
+        },
+      });
+    });
+
+    it('should convert the field types of Compound type', () => {
```

Comment:
The test might seem over-complicated but it makes sure that all types are converted (even deeply nested ones).

It cannot happen in `Array` in `VLen` types as the base is a `HsdsBaseType` in these cases.